File: java-spring/src/main/java/com/example/javaspring/db/UserModel.java
package com.example.javaspring.db;

import com.example.javaspring.util.GetColor;
import lombok.Data;

import java.util.ArrayList;
import java.util.Collections;

@Data // Lombok 어노테이션. getter/setter, toString 등의 메서드 자동 생성
public class UserModel { // model은 Front로 보낼 응답에 사용될 객체의 형식

    Long id;
    String name;
    Long age;
    String favColor; // 실제 DB 테이블에 저장된 User 엔티티의 형식과 달라도 됨

    public UserModel(UserEntity e){
        this(e, null, null); // this() : 내부의 다른 생성자 호출
    }

    public UserModel(
            UserEntity e,
            String colorInput,
            String randomColor
    ) {
        if (colorInput == null) {
            if (randomColor == null)
                randomColor = GetColor.randomColor(new ArrayList<String>(Collections.emptyList()));
            colorInput = randomColor;
        }
        this.id = e.id;
        this.name = e.name;
        this.age = e.age;
        this.favColor = colorInput;
    }
}

File: kotlin-spring/src/main/kotlin/com/example/kotlinspring/db/UserModel.kt
package com.example.kotlinspring.db

import com.example.kotlinspring.util.GetColor

// model은 Front로 보낼 응답에 사용될 객체의 형식
data class UserModel (
    val id: Long,
    val name: String,
    val age: Long,
    val favColor: String, // 실제 DB 테이블에 저장된 User 엔티티의 형식과 달라도 됨
){
    companion object {
        fun of(
            e: UserEntity,
            colorInput: String? = null,
            randomColor: String = GetColor.randomColor(listOf())
        ): UserModel =
            UserModel(
                id = e.id,
                name = e.name,
                age = e.age,
                favColor = colorInput ?: randomColor
            )
    }
}